libraries/browser-tracker-core/src/tracker/id_cookie.ts

```diff
@@ -0,0 +1,356 @@
+    tmpContainer[sessionIdIndex] = uuid();
+  }
+  if (!tmpContainer[previousSessionIdIndex] || tmpContainer[previousSessionIdIndex] === 'undefined') {
+    // previous session id
+    tmpContainer[previousSessionIdIndex] = '';
+  }
+  if (!tmpContainer[firstEventIdIndex] || tmpContainer[firstEventIdIndex] === 'undefined') {
+    // firstEventId - blank meaning no previous event
+    tmpContainer[firstEventIdIndex] = '';
+  }
+  if (!tmpContainer[firstEventTsInMsIndex] || tmpContainer[firstEventTsInMsIndex] === 'undefined') {
+    // firstEventTs - blank meaning no previous event
+    tmpContainer[firstEventTsInMsIndex] = '';
+  }
+  if (!tmpContainer[eventIndexIndex] || tmpContainer[eventIndexIndex] === 'undefined') {
+    // eventIndex – 0 = no previous event
+    tmpContainer[eventIndexIndex] = 0;
+  }
+
+  const parsed: ParsedIdCookie = [
+    tmpContainer[0] as string,
+    tmpContainer[1] as string,
+    parseInt(tmpContainer[2] as string),
+    parseInt(tmpContainer[3] as string),
+    parseInt(tmpContainer[4] as string),
+    tmpContainer[5] ? parseInt(tmpContainer[5] as string) : (tmpContainer[5] as string),
+    tmpContainer[6] as string,
+    tmpContainer[7] as string,
+    tmpContainer[8] as string,
+    tmpContainer[9] ? parseInt(tmpContainer[9] as string) : (tmpContainer[9] as string),
+    parseInt(tmpContainer[10] as string),
+  ];
+  return parsed;
+}
+
+/**
+ * Initializes the domain user ID if not already present in the cookie. Sets an empty string if anonymous tracking.
+ *
+ * @param idCookie Parsed cookie
+ * @param configAnonymousTracking Whether anonymous tracking is enabled
+ * @returns Domain user ID
+ */
+export function initializeDomainUserId(idCookie: ParsedIdCookie, configAnonymousTracking: boolean) {
+  let domainUserId;
+  if (idCookie[domainUserIdIndex]) {
+    domainUserId = idCookie[domainUserIdIndex];
+  } else if (!configAnonymousTracking) {
+    domainUserId = uuid();
+    idCookie[domainUserIdIndex] = domainUserId;
+  } else {
+    domainUserId = '';
+    idCookie[domainUserIdIndex] = domainUserId;
+  }
+  return domainUserId;
+}
+
+/**
+ * Starts a new session with a new ID.
+ * Sets the previous session, last visit timestamp, and increments visit count if cookies enabled.
+ * First event references are reset and will be updated in `updateFirstEventInIdCookie`.
+ *
+ * @param idCookie Parsed cookie
+ * @param memorizedVisitCount Visit count to be used if cookies not enabled
+ * @returns New session ID
+ */
+export function startNewIdCookieSession(idCookie: ParsedIdCookie, memorizedVisitCount: number = 1) {
+  // If cookies are enabled, base visit count and session ID on the cookies
+  if (cookiesEnabledInIdCookie(idCookie)) {
+    // Store the previous session ID
+    idCookie[previousSessionIdIndex] = idCookie[sessionIdIndex];
+    // Set lastVisitTs to currentVisitTs
+    idCookie[lastVisitTsIndex] = idCookie[nowTsIndex];
+    // Increment the session ID
+    (idCookie[visitCountIndex] as number)++;
```

Comment:
Isn't this already a number?

```suggestion
    idCookie[visitCountIndex]++;
```